xclim/indices/generic.py

```diff
@@ -594,3 +594,56 @@ def extreme_temperature_range(
     u = str2pint(low_data.units)
     out.attrs["units"] = pint2cfunits(u - u)
     return out
+
+
+def aggregate_between_dates(data, start, end, op="sum", freq=None):
+    """Aggregate the data over a period between start and end dates and apply the operator on the aggregated data.
+
+    Parameters
+    ----------
+    data : xr.DataArray
+      Data to aggregate between start and end dates.
+    start
+      Start dates for the aggregation periods.
+    end
+      End dates for the aggregation periods.
+    op : str
```

Comment:
```suggestion
    op : {'min', 'max', 'sum', 'mean', 'std'}
```
A user could try with other "operators", but those are the ones we usually list for that case. Anyway, it's only an annotation.